app/controllers/vocabs_controller.rb

```diff
@@ -0,0 +1,52 @@
+class VocabsController < ApplicationController
+
+  def index
+    @vocabs = Vocab.all
+  end
+
+  def create
+
+    @vocab = Vocab.new(vocab_params)
+
+    if @vocab.save
+      #render success in Jbuilder
+    else
+      render json: { message: "400 Bad Request" }, status: :bad_request
+    end
+
+  end
+
+  def update
+    @vocab = Vocab.find_by_id(params[:id])
+
+    if @vocab.nil?
+      render json: { message: "Cannot find your vocab" }, status: :not_found
+    else
+      @vocab.update(vocab_params)
```

Comment:
`@vocab.update(vocab_params)` may return true or false, depending if we can save the vocab to the database, you should check on that and return an error if it returns false